cmd/user/repository.go

```diff
@@ -0,0 +1,197 @@
+package user
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/neidersalgado/go-camp-grpc/cmd/user/pb"
+)
+
+type MySQLUserRepository struct {
+	ConnectionClient *sql.DB
+	//TODO slice the contention with permissions read an write
+}
+
+const (
+	querySaveUser    = `INSERT INTO user(name,pwdhash,age,aditional_information,email) values (?,?,?,?,?);`
+	querySaveParents = `INSERT INTO parent(parent, son) (?,?)`
+	queryDeleteUser  = `DELETE FROM user WHERE email = ?`
+	queryGetUser     = `SELECT Id, name, pwdhash, age, aditional_information FROM user WHERE email = ?`
+	queryGetUsers    = `SELECT Id, email, name, pwdhash, age, aditional_information FROM user`
+	queryUpdateUSer  = `UPDATE user SET name= ?, pwdhash = ?, age =?, aditional_information =? WHERE email = ? `
+)
+
+func NewMySQLUserRepository(connection *sql.DB) *MySQLUserRepository {
+	return &MySQLUserRepository{
+		ConnectionClient: connection,
+	}
+}
+
+func (r *MySQLUserRepository) Create(user pb.UserRequest) error {
+	stmtSaveUser, err := r.ConnectionClient.Prepare(querySaveUser)
+
+	if err != nil {
+		return fmt.Errorf(
+			fmt.Sprintf("Connetion Error, Couldn't save User With ID: %s in database, Error: %v", user.Email, err.Error()),
+		)
+	}
+	fmt.Println(user.PwdHash)
```

Comment:
Please do not log any sensitive information. This will be red flag in a PR with the client